### PR TITLE
positioningContainer to include changes in v8

### PR DESCRIPTION
**Pull request by sbarrameda**
## Current Behavior

Target moves on scroll; if coachmark is in dialog, and dialog body string is too long, the target of the coachmark moves
## New Behavior
behavior in v8. target does not move. coachmark positions to correct target

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by sbarrameda** [comment]
I ported changes from v8 Coachmark/PositioningContainer to this version

File: packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.styles.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { memoizeFunction } from '../../../Utilities';
-import { mergeStyleSets, focusClear, IStyle, HighContrastSelector } from '../../../Styling';
+import { mergeStyleSets, focusClear, HighContrastSelector } from '../../../Styling';
+import type { IStyle } from '../../../Styling';
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.styles.ts:3]: This one needs to be reverted--v7 doesn't support type imports
```suggestion
import { mergeStyleSets, focusClear, IStyle, HighContrastSelector } from '../../../Styling';
```

File: packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.tsx [hunk 2]
```diff
@@ -1,438 +1,395 @@
...
-  IPosition,
-  RectangleEdge,
-} from '../../../utilities/positioning';
+import { css, elementContains, focusFirstChild, EventGroup, getPropsWithDefaults } from '../../../Utilities';
+
+import { getMaxHeight, positionElement, RectangleEdge } from '../../../Positioning';
 
 import { AnimationClassNames, mergeStyles } from '../../../Styling';
+import { useMergedRefs, useAsync, useTarget } from '@fluentui/react-hooks';
+import type { IPositioningContainerProps } from './PositioningContainer.types';
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.tsx:14]: These `import type` changes also need to be reverted. (Same in all other files.)

**Review by ecraig12345 — CHANGES_REQUESTED** [review]
Thanks for the contribution, but all the changes to use `import type` syntax need to be reverted to avoid causing compatibility issues for consumers of v8. (This syntax was only added in typescript 3.8, and our consumer min bar for v7 is 3.5.)

**Pull request closed without merging**